### DISCUSSION TITLE
Failed to initialize axoverlay: GDBus.Error:com.axis.Graphics2.Error.Unauthorized: Unauthorized Access

### DISCUSSION THREAD

**Discussion by rywager** [AxisCommunications/acap-native-sdk-examples#230] [General] [answered]
I am on a AXIS P3268-LVE Dome Camera running Current version: 11.9.60

I am using pretty much exactly the examples i see everywhere and i get the error in subject line whenever i start up.

Anyone have insight as to what "Failed to initialize axoverlay: GDBus.Error:com.axis.Graphics2.Error.Unauthorized: Unauthorized Access" means?

`static void
render_overlay_cb(gpointer render_context, gint id,
                   struct axoverlay_stream_data *stream,
                   enum axoverlay_position_type postype, gfloat overlay_x,
                   gfloat overlay_y, gint overlay_width, gint overlay_height,
                   gpointer user_data)
{

    GError *error = NULL;
    // Get maximum camera resolution
    gint camera_width = axoverlay_get_max_resolution_width(1, &error);
    gint camera_height = axoverlay_get_max_resolution_height(1, &error);
    if (error) {
        syslog(LOG_ERR, "Failed to get camera resolution: %s", error->message);
        g_error_free(error);
        return 1;
    }

    static gdouble pie_angle = 0.0;

    cairo_t *cr = render_context;
    gdouble x = camera_width/2;
    gdouble y = camera_height/2;
    gdouble radius = PIE_RADIUS;
    gdouble angle1 = 0;
    gdouble angle2 = pie_angle*(M_PI/180.0);

    /* Clear background */
    cairo_set_source_rgba(cr, 0.0, 0.0, 0.0, 0.0);
    cairo_set_operator(cr, CAIRO_OPERATOR_SOURCE);
    cairo_rectangle(cr, 0, 0, camera_width, camera_height);
    cairo_fill(cr);

    /* Draw a filled pie shape */
    cairo_set_source_rgba(cr, 1.0, 0.0, 0.0, 1.0);
    cairo_set_operator(cr, CAIRO_OPERATOR_SOURCE);
    cairo_set_line_width(cr, 5.0);
    cairo_arc(cr, x, y, radius, angle1, angle2);
    cairo_line_to(cr, x, y);
    cairo_line_to(cr, x+PIE_RADIUS, y);
    cairo_fill(cr);
}

int main() {

    GError *error = NULL;

    axoverlay_reload_streams(&error);
    if (error) {
        syslog(LOG_ERR, "Failed to reload streams: %s", error->message);
        g_error_free(error);
        return 1;
    }

    if (!axoverlay_is_backend_supported(AXOVERLAY_CAIRO_IMAGE_BACKEND)) {
        syslog(LOG_ERR, "AXOVERLAY_CAIRO_IMAGE_BACKEND is not supported");
        return 1;
    } else {
        syslog(LOG_INFO, "AXOVERLAY_CAIRO_IMAGE_BACKEND is supported");
    }

    syslog(LOG_INFO, "Starting up...");
    /* Initialize the library */
    struct axoverlay_settings settings;
    axoverlay_init_axoverlay_settings(&settings);
    settings.render_callback = render_overlay_cb;
    settings.adjustment_callback = NULL;
    settings.select_callback = NULL;
    settings.backend = AXOVERLAY_CAIRO_IMAGE_BACKEND;
    axoverlay_init(&settings, &error);
    if (error != NULL) {
        syslog(LOG_INFO,"Failed to initialize axoverlay: %s", error->message);
        g_error_free(error);
        return 1;
    }`

**Comment by Cacsjep** [comment, marked as answer]
U forget Manifest entry for dbus Ressource right?

**Reply by pataxis** [reply]
Yes, as @Cacsjep  says, to use axoverlay you need to add the the D-Bus resources to the [manifest.json](https://github.com/AxisCommunications/acap-native-sdk-examples/blob/main/axoverlay/app/manifest.json#L5) when using a [dynamic user](https://axiscommunications.github.io/acap-documentation/docs/develop/application-project-structure#dynamic-user).

**Reply by rywager** [reply]
That absolutely did it, thank you as always @Cacsjep !

`{
    "schemaVersion": "1.6.0",
    "resources": {
        "dbus": {
            "requiredMethods": [
                "com.axis.Graphics2.*",
                "com.axis.Overlay2.*"
            ]
        }
    },
    "acapPackageConf": {`
